### ISSUE TITLE
Missing dependency: which python version did you use?

### ISSUE THREAD

**Issue by a1ultima** [llSourcell/visualize_dataset_demo#2]
Missing dependency: which python version did you use?

Hi Siraj,

Thanks for listing dependent Python libraries, but for us wondering which Python version to create a virtualenv with (e.g. if we have loads of Python versions) can I ask which you used? You did not put a shebang on the top of your live demo. 

Best,

Andy

**Comment by a1ultima** [comment]
Hi again,

It seems the data_visualization.py does not yet work, even with locally downloaded pml-training.py found in: http://rstudio-pubs-static.s3.amazonaws.com/19668_2a08e88c36ab4b47876a589bb1d61c37.html﻿

Since I am currently using python 2.7.3 it is hard to attribute the data_visualization.py not working to factors other than python version, e.g. that the code actually doesn't work... Anyway, a reason to suspect the python version is not the cause of the data_visualization.py "just doesn't work" is that I cannot see at which point any training is done on the data set in the code itself. Sure you use sklearn to split data into test and training sets but I cannot see any gradient descent going on.

I will soon use python >3.3, and make changes until it does work (e.g. slicing indices for getting class labels); there's a high chance the [pml-training,py data I tried to find](http://rstudio-pubs-static.s3.amazonaws.com/19668_2a08e88c36ab4b47876a589bb1d61c37.html﻿) is not the same as discussed in [your youtube video](https://www.youtube.com/watch?v=yQsOFWqpjkE&t=323s).
